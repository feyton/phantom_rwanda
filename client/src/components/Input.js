import React, { forwardRef } from 'react';

const Input = forwardRef(
  (
    {
      type,
      placeholder,
      styles,
      onChange,
      name,
      htmlFor,
      labelName,
      defaultValue,
      wrapperStyles,
      id,
      disabled,
      pattern,
      required
    },
    ref
  ) => {
    return (
      <div className={`${wrapperStyles} mb-2 block font-raleway`}>
        <label
          htmlFor={htmlFor}
          className="block text-grey-darker text-sm font-bold font-rale mb-2"
        >
          {labelName}
        </label>
        <input
          className={`${styles} appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden`}
          type={type}
          placeholder={placeholder}
          defaultValue={defaultValue}
          onChange={onChange}
          disabled={disabled}
          name={name}
          ref={ref}
          id={id}
          required={required || false}
          pattern={pattern}
        />
      </div>
    );
  }
);

export default Input;
