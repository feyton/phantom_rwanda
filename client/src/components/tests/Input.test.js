import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input.js';

describe('Input', () => {
  it('should render Input', () => {
    const input = renderer.create(<Input />).toJSON();
    expect(input).toMatchSnapshot();
  });
});
