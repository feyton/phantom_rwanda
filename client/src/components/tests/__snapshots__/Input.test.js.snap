// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input 1`] = `
<div
  className="undefined mb-2 block font-raleway"
>
  <label
    className="block text-grey-darker text-sm font-bold font-rale mb-2"
  />
  <input
    className="undefined appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
    required={false}
  />
</div>
`;
